puma/apps/android/snapchat/snapchat.py

```diff
@@ -1,175 +1,178 @@
-@supported_version("12.90.0.46")
-class SnapchatActions(AndroidAppiumActions):
-    def __init__(self,
-                 device_udid,
-                 desired_capabilities: Dict[str, str] = None,
-                 implicit_wait=1,
-                 appium_server='http://localhost:4723'):
+    :param driver: The PumaDriver instance used to interact with the application.
+    :param conversation: The name of the conversation to navigate to.
+    """
+
+    driver.get_elements(CHAT_CONVERSATION.format(conversation=conversation))[-1].click()
+    driver.click(FRIEND_ACTION)
+
+class SnapchatChatState(SimpleState, ContextualState):
+    """
+    A state representing a chat screen in the application.
+
+    This class extends both SimpleState and ContextualState to represent a chat screen
+    and validate its context based on the conversation name.
+    """
+
+    def __init__(self, parent_state):
         """
-        Class with an API for Snapchat Android using Appium. Can be used with an emulator or real device attached to the computer.
+        Initializes the ChatState with a parent state.
+
+        :param parent_state: The parent state of this chat state.
         """
-        AndroidAppiumActions.__init__(self,
-                                      device_udid,
-                                      SNAPCHAT_PACKAGE,
-                                      desired_capabilities=desired_capabilities,
-                                      implicit_wait=implicit_wait,
-                                      appium_server=appium_server)
+        super().__init__(xpaths=[CONVERSATION_TITLE, CHAT_INPUT],
+                         parent_state=parent_state)
 
-    def _currently_at_homescreen(self) -> bool:
-        return self.is_present(
-            '//android.widget.LinearLayout[@resource-id="com.snapchat.android:id/ngs_navigation_bar"]') \
-            and not self.is_present('//*[@text="View Profile"]')
+    def validate_context(self, driver: PumaDriver, conversation: str = None) -> bool:
+        """
+        Validates the context of the chat state.
 
-    def _currently_in_conversation_overview(self) -> bool:
-        return self.is_present(
-            '//android.widget.TextView[@resource-id="com.snapchat.android:id/hova_page_title" and @text="Chat"]')
+        This method checks if the current chat screen matches the expected conversation name.
 
-    def _currently_on_top_of_conversation_overview(self) -> bool:
+        :param driver: The PumaDriver instance used to interact with the application.
+        :param conversation: The name of the conversation to validate against.
+        :return: True if the context is valid, False otherwise.
         """
-        Check if the current position is at the top of the conversation view.
+        if not conversation:
+            return True
+
+        content_desc = driver.get_element(CONVERSATION_TITLE).get_attribute('text')
+        return conversation in content_desc
```

Comment:
Replace this with the XPATH `contains()` function. I would also make this case-insensitive.